web/vibe/web/common.d

```diff
@@ -479,6 +492,39 @@ package struct MethodAttribute
 	alias data this;
 }
 
+///
+public struct ResultSerializer (alias ST, alias DT, string ContentType)
+{
+	enum contentType = ContentType;
+	alias serialize = ST;
+	alias deserialize = DT;
+}
+
+alias resultSerializer(Args...) = ResultSerializer!(Args);
+
+///
+package alias DefaultSerializerT (FuncRetT) =
+	ResultSerializer!(
+		function (ref output_stream, ref value){
+			output_stream ~= serializeToJsonString(value);
+		},
+		function (input_stream){
```

Comment:
I was unable to extract the actual stream object from the ProxyInterface as it currently requires knowing the type of the stream object compile time, but it is only available in runtime. As a result of that currently the type of the 'input stream' is ProxyInterface!InputStream. 

StreamInputRange also didn't work thanks to the ProxyInterface

Reply:
I didn't realize that both of those were the case - `StreamInputRange` only accepting `InputRange` and the JSON parser code requiring a forward range (for `skipNumber` only as it seems). I'll see what I can do there.